legacy/simple_classif_training.py

```diff
@@ -0,0 +1,175 @@
+
+
+class SklearnDataModule(LightningDataModule):
+
+    def __init__(self, sklearn_dataset, x_type, y_type, batch_size: int = 128):
+        super().__init__()
+        self.batch_size = batch_size
+        self._x, self._y = sklearn_dataset
+        self._split_data()
+        self._x_type = x_type
+        self._y_type = y_type
+
+    def _split_data(self):
+        self.x_train, self.x_test, self.y_train, self.y_test = train_test_split(
+            self._x, self._y, test_size=0.20, random_state=42
+        )
+        self.x_train, self.x_valid, self.y_train, self.y_valid = train_test_split(
+            self.x_train, self.y_train, test_size=0.40, random_state=42
+        )
+
+    def train_dataloader(self):
+        return DataLoader(
+            SklearnDataset(self.x_train, self.y_train, self._x_type, self._y_type),
+            shuffle=True,
+            batch_size=self.batch_size
+        )
+
+    def val_dataloader(self):
+        return DataLoader(
+            SklearnDataset(self.x_valid, self.y_valid, self._x_type, self._y_type), batch_size=self.batch_size
+        )
+
+    def test_dataloader(self):
+        return DataLoader(
+            SklearnDataset(self.x_test, self.y_test, self._x_type, self._y_type), batch_size=self.batch_size
+        )
+
+
+class ClassifDataModule(SklearnDataModule):
+
+    def __init__(self, num_features=24, length=6000, num_classes=3, batch_size=128):
+        data = make_classification(
+            n_samples=length,
+            n_features=num_features,
+            n_classes=num_classes,
+            n_clusters_per_class=2,
+            n_informative=int(num_features / num_classes),
+            random_state=42,
+        )
+        super().__init__(data, x_type=torch.float32, y_type=torch.long, batch_size=batch_size)
+
+
+class ClassificationModel(LightningModule):
+
+    def __init__(self, num_features=24, num_classes=3, lr=0.01):
+        super().__init__()
+        self.save_hyperparameters()
+
+        self.lr = lr
+        for i in range(3):
+            setattr(self, f"layer_{i}", nn.Linear(num_features, num_features))
+            setattr(self, f"layer_{i}a", torch.nn.ReLU())
+        setattr(self, "layer_end", nn.Linear(num_features, num_classes))
+
+        self.train_acc = Accuracy()
+        self.valid_acc = Accuracy()
+        self.test_acc = Accuracy()
+
+    def forward(self, x):
+        x = self.layer_0(x)
+        x = self.layer_0a(x)
+        x = self.layer_1(x)
+        x = self.layer_1a(x)
+        x = self.layer_2(x)
+        x = self.layer_2a(x)
+        x = self.layer_end(x)
+        logits = F.softmax(x, dim=1)
+        return logits
+
+    def configure_optimizers(self):
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.lr)
+        return [optimizer], []
+
+    def training_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self.forward(x)
+        loss = F.cross_entropy(logits, y)
+        self.log('train_loss', loss, prog_bar=True)
```

Comment:
Hey @Borda, quick questions to improve this test for stability.

Should we check the loss values to ensure consistency ?
Should we test manual optimization ?
Should we test Callback states restoration ?

Reply:
> Should we check the loss values to ensure consistency?

now we check `test_loss` so you suggest testing all three losses - train/val/test?

> Should we test manual optimization?

do you mean to add another model with manual optimization?

> Should we test Callback states restoration?

That would be good but it requires to have different source code per past version as not all callbacks were present since 1.0 yet... so lets make it a separate PR

Reply:
1. Should we check the loss values to ensure consistency?

Here is what I meant.

```
def test_legacy()

    target_losses = [1.2, 1....] # recorded once and used to ensure all models converges in the same way after each release

    for chk in checkpoints:
        seed_everything(42)
        losses = evaluate(chk)
        assert losses == previous_losses
```

2. do you mean to add another model with manual optimization? Yes.

3. Yes, we can start from the lowest version properly supporting the callback states. Definitely within another PR.

Reply:
> 1. Should we check the loss values to ensure consistency?
> 
> Here is what I meant.
> 
> ```
> def test_legacy()
> 
>     target_losses = [1.2, 1....] # recorded once and used to ensure all models converges in the same way after each release
> 
>     for chk in checkpoints:
>         seed_everything(42)
>         losses = evaluate(chk)
>         assert losses == previous_losses
> ```

on, that would need to save the values so lets make it another PR
 
> 2. do you mean to add another model with manual optimization? Yes.

Would need to change the logic for finding ckpt and name then, so another PR

> 3. Yes, we can start from the lowest version properly supporting the callback states. Definitely within another PR.

another PR :rabbit: